.github/workflows/master.yml

```diff
@@ -170,6 +170,61 @@ jobs:
             **/kyuubi-jdbc-engine.log*
             **/kyuubi-hive-sql-engine.log*
 
+  spark-connector-cross-version-test:
+    name: Spark Connector Cross Version Test
+    runs-on: ubuntu-22.04
+    strategy:
+      fail-fast: false
+      max-parallel: 1
+      matrix:
+        java:
+          - 17
+        scala:
+          - '2.12'
+          - '2.13'
+        spark-compile:
+            - '3.5'
+        spark-runtime:
+            - '3.4'
+            - '3.3'
+        comment: [ "normal" ]
+    env:
+      SPARK_LOCAL_IP: localhost
+    steps:
+      - uses: actions/checkout@v4
+      - name: Free up disk space
+        run: ./.github/scripts/free_disk_space.sh
+      - name: Tune Runner VM
+        uses: ./.github/actions/tune-runner-vm
+      - name: Setup JDK ${{ matrix.java }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.java }}
+          cache: 'maven'
+          check-latest: false
+      - name: Setup Maven
+        uses: ./.github/actions/setup-maven
+      - name: Build Kyuubi Spark Hive Connector with Spark-${{ matrix.spark-compile }}
+        run: |
+          ./build/mvn clean install ${MVN_OPT} -pl extensions/spark/kyuubi-spark-connector-hive -am \
```

Comment:
The TPC-DS/H connectors should be compatible too, can we add them to the x-version test too?

Reply:
Sure. I'll do it in another PR.